### PR TITLE
fix: saving fixtures locally (for development)

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves https://github.com/Arize-ai/phoenix/issues/650

### CODE REVIEW

File: src/phoenix/datasets/fixtures.py [hunk 1]
```diff
@@ -15,13 +20,22 @@
 class Fixture:
     name: str
     description: str
-    primary_dataset_url: str
-    reference_dataset_url: Optional[str]
+    prefix: str
+    primary: str
+    reference: Optional[str]
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:25]:
```suggestion
    primary_file_name: str
    reference_file_name: Optional[str]
```

File: src/phoenix/datasets/fixtures.py [hunk 2]
```diff
@@ -414,8 +374,59 @@ def load_example(use_case: str) -> ExampleDatasets:
...
     print(f"📥 Loaded {use_case} example datasets.")
     print("ℹ️ About this use-case:")
     print(fixture.description)
     return ExampleDatasets(primary=primary_dataset, reference=reference_dataset)
+
+
+class Metadata(NamedTuple):
+    path: str
+    mediaLink: str
+    md5Hash: str
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:387]: this is a file-level hash to re-download? Nifty if so.

File: src/phoenix/datasets/fixtures.py [hunk 3]
```diff
@@ -414,8 +374,59 @@ def load_example(use_case: str) -> ExampleDatasets:
...
+
+    def save_media(self, location: Path) -> Path:
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:389]: nit: media makes me think video / audio. Something like data or artifacts might be more appropriate

File: src/phoenix/datasets/fixtures.py [hunk 4]
```diff
@@ -414,8 +374,59 @@ def load_example(use_case: str) -> ExampleDatasets:
...
+        data_file = location / self.path
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:390]:
```suggestion
        data_files_path = location / self.path
```

File: src/phoenix/datasets/fixtures.py [hunk 5]
```diff
@@ -414,8 +374,59 @@ def load_example(use_case: str) -> ExampleDatasets:
...
+        request.urlretrieve(self.mediaLink, data_file)
+        with open(md5_file, "w") as f:
+            f.write(self.md5Hash)
+        return data_file
+
+
+class GCS(NamedTuple):
+    host: str = "https://storage.googleapis.com/"
+    bucket: str = "arize-assets"
+    prefix: str = "phoenix/datasets/"
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:406]: Can't re-use the constants here?

File: src/phoenix/datasets/fixtures.py [hunk 6]
```diff
@@ -414,8 +374,59 @@ def load_example(use_case: str) -> ExampleDatasets:
...
+            with open(md5_file, "r") as f:
+                if f.readline() == self.md5Hash:
+                    return data_file
+        request.urlretrieve(self.mediaLink, data_file)
+        with open(md5_file, "w") as f:
+            f.write(self.md5Hash)
+        return data_file
+
+
+class GCS(NamedTuple):
```
Comment (mikeldking) [inline src/phoenix/datasets/fixtures.py:403]:
```suggestion
class GCSAsset(NamedTuple):
```
Better to be clear it's not just GCS :)

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/datasets/fixtures.py (hunk 2 above)
Comment (RogerHYang) [reply src/phoenix/datasets/fixtures.py:387]: yup. this is file-level binary md5 encoded as base64

File: src/phoenix/datasets/fixtures.py (hunk 5 above)
Comment (RogerHYang) [reply src/phoenix/datasets/fixtures.py:406]: they may have the same value, but are semantically distinct (i.e. one is for gcs and the other is for the local file system), so i kept them separate

**Pull request merged**